tests/functional/c/comparison_with_callable_typing_constants.py

```diff
@@ -0,0 +1,19 @@
+"""
+Typing constants are actually implemented as functions, but they
+raise when called, so Pylint uses that to avoid false positives for
+comparison-with-callable.
+"""
+from typing import Any, Optional
+
+
+def check_any(type_) -> bool:
+    """See https://github.com/PyCQA/pylint/issues/5557"""
+    return type_ == Any
+
+
+def check_optional(type_) -> bool:
+    """
+    Unlike Any, Optional does not raise in its body.
```

Comment:
```suggestion
    """Unlike Any, Optional does not raise in its body.
```

Good job catching this as well!

Reply:
Fun fact: all of the typing constants raise first via the decorator rather than via their body, but when I saw that, I realized we ought to take care of detecting raises in the body, since that's easier.